spicy/zeek.spicy

```diff
@@ -43,36 +43,45 @@ public function flip_roles() : void &cxxname="spicy::zeek::rt::flip_roles";
 ## Returns the number of packets seen so far on the current side of the current connection.
 public function number_packets() : uint64 &cxxname="spicy::zeek::rt::number_packets";
 
+## Opaque handle to a protocol analyzer.
+public type ProtocolHandle = __library_type("spicy::zeek::rt::ProtocolHandle");
+
 ## Adds a Zeek-side child protocol analyzer to the current connection.
 ##
 ## analyzer: type of analyzer to instantiate, specified through its Zeek-side
 ## name (similar to what Zeek's signature action `enable` takes); if not
 ## specified, Zeek will perform its usual dynamic protocol detection to figure
 ## out how to parse the data (the latter will work only for TCP protocols, though.)
-public function protocol_begin(analyzer: optional<string> = Null) : void &cxxname="spicy::zeek::rt::protocol_begin";
+##
+## Returns identifier of the child analyzer.
+public function protocol_begin(analyzer: optional<string> = Null) : optional<ProtocolHandle> &cxxname="spicy::zeek::rt::protocol_begin";
```

Comment:
Ideally we'd have integration tests for this explicit management of protocol handles.